core/src/main/java/org/apache/accumulo/core/metadata/TabletFile.java

```diff
@@ -55,27 +57,54 @@ public class TabletFile implements Comparable<TabletFile> {
    */
   public TabletFile(Path metaPath) {
     this.metaPath = Objects.requireNonNull(metaPath);
-    String errorMsg = "Missing or invalid part of tablet file metadata entry: " + metaPath;
+    final String errorMsg = "Missing or invalid part of tablet file metadata entry: " + metaPath;
     log.trace("Parsing TabletFile from {}", metaPath);
 
-    // use Path object to step backwards from the filename through all the parts
-    this.fileName = metaPath.getName();
-    ValidationUtil.validateFileName(fileName);
+    // File name construct: <volume>/<tablePath>/<tableId>/<tablet>/<file>
+    // Example: hdfs://namenode:9020/accumulo/tables/1/default_tablet/F00001.rf
+    final String path = this.metaPath.toUri().getPath();
+    final String[] parts = path.split("/");
+    final int numParts = parts.length;
 
-    Path tabletDirPath = Objects.requireNonNull(metaPath.getParent(), errorMsg);
+    if (numParts < 4) {
+      throw new IllegalArgumentException(errorMsg);
+    }
 
-    Path tableIdPath = Objects.requireNonNull(tabletDirPath.getParent(), errorMsg);
-    var id = tableIdPath.getName();
+    // step backwards from the filename through all the parts
+    String file = null;
+    String tabletDirectory = null;
+    String tableId = null;
+    for (int i : IntStream.of(numParts - 1, numParts - 2, numParts - 3, numParts - 4).toArray()) {
+      String tmp = Objects.requireNonNull(parts[i], errorMsg);
+      if (i == numParts - 1) {
+        file = tmp;
+        ValidationUtil.validateFileName(file);
+      } else if (i == numParts - 2) {
+        tabletDirectory = tmp;
+      } else if (i == numParts - 3) {
+        tableId = tmp;
+      } else if (i == numParts - 4) {
+        String tablesPath = "/" + tmp;
+        Preconditions.checkArgument(tablesPath.equals(HDFS_TABLES_DIR),
+            "tables path is not " + HDFS_TABLES_DIR + ", is " + tablesPath);
+      }
+    }
```

Comment:
Could this loop be replaced w/ the following?  Also thinking we do not need the non-null checks as I do not think the String split function will ever return nulls.

```suggestion    
    file = parts[numParts -1];
    ValidationUtil.validateFileName(file);
    tabletDirectory = parts[numParts - 2];
    tableId = parts[numParts - 3];
    String tablesPath = parts[numParts - 4];
    // HDFS_TABLES_DIR_2 is a precomputed version of the constant w/o a leading slash
    Preconditions.checkArgument(tablesPath.equals(HDFS_TABLES_DIR_2),
            "tables path is not " + HDFS_TABLES_DIR_2 + ", is " + tablesPath);
```

Reply:
Good suggestion, I think I was just following a habit. Implemented in 0ad1124.